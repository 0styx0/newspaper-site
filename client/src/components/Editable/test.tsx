import * as React from 'react';
import * as renderer from 'react-test-renderer';
import casual from '../../tests/casual.data';
import EditableContainer from './container';
import { mount } from 'enzyme';
import * as sinon from 'sinon';

document.queryCommandSupported = () => true;

describe('<EditableContainer>', () => {

    function setup(onSubmit: Function) {

        return mount(
            <EditableContainer
              children={<div />}
              canEdit={true}
              buttons="all"
              onSubmit={onSubmit}
            />
        );
    }

    describe('snapshots', () => {

        /**
         * Takes snapshots
         */
        function snap(buttons: 'all' | 'basic' | 'none', canEdit: boolean = true) {

            const tree = renderer.create(
                <EditableContainer
                    children={<p>Hi, this is a test</p>}
                    canEdit={canEdit}
                    buttons={buttons}
                    onSubmit={casual.function}
                />
            ).toJSON();

            expect(tree).toMatchSnapshot();
        }

        describe('render buttons correctly when set to', () => {

            it('all', () => snap('all'));

            it('basic', () => snap('basic'));

            it('none', () => snap('none'));
        });

        describe('render correctly when canEdit is set to', () => {

            test('true', () => snap('all', true));

            test('false', () => snap('all', false));
        });
    });

    describe('onSubmit', () => {

        it('is called when submit button is clicked', () => {

            const spy = sinon.spy();

            const wrapper = setup(spy);

            wrapper.find('button').last().simulate('click');

            expect(spy.called).toBeTruthy();
        });
    });
});