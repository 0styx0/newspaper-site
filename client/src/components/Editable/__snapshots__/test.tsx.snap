// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditableContainer> snapshots render buttons correctly when set to all 1`] = `
<div>
  <div
    id="buttonContainer"
  >
    <button
      className="bold"
      onClick={[Function]}
      type="button"
    >
      bold
    </button>
    <button
      className="italic"
      onClick={[Function]}
      type="button"
    >
      italic
    </button>
    <button
      className="underline"
      onClick={[Function]}
      type="button"
    >
      underline
    </button>
    <button
      className="heading"
      onClick={[Function]}
      type="button"
    >
      heading
    </button>
    <button
      className="fontSize"
      onClick={[Function]}
      type="button"
    >
      fontSize
    </button>
    <button
      className="insertOrderedList"
      onClick={[Function]}
      type="button"
    >
      insertOrderedList
    </button>
    <button
      className="insertUnorderedList"
      onClick={[Function]}
      type="button"
    >
      insertUnorderedList
    </button>
    <button
      className="justifyFull"
      onClick={[Function]}
      type="button"
    >
      justifyFull
    </button>
    <button
      className="justifyLeft"
      onClick={[Function]}
      type="button"
    >
      justifyLeft
    </button>
    <button
      className="justifyRight"
      onClick={[Function]}
      type="button"
    >
      justifyRight
    </button>
    <button
      className="justifyCenter"
      onClick={[Function]}
      type="button"
    >
      justifyCenter
    </button>
    <button
      className="strikeThrough"
      onClick={[Function]}
      type="button"
    >
      strikeThrough
    </button>
    <button
      className="subscript"
      onClick={[Function]}
      type="button"
    >
      subscript
    </button>
    <button
      className="superscript"
      onClick={[Function]}
      type="button"
    >
      superscript
    </button>
    <button
      className="insertImage"
      onClick={[Function]}
      type="button"
    >
      insertImage
    </button>
    <button
      className="createLink"
      onClick={[Function]}
      type="button"
    >
      createLink
    </button>
    <button
      className="unlink"
      onClick={[Function]}
      type="button"
    >
      unlink
    </button>
    <button
      className="hideFromPreview"
      onClick={[Function]}
      type="button"
    >
      hideFromPreview
    </button>
    <br />
    <button
      onClick={undefined}
      type="button"
    >
      Submit
    </button>
  </div>
  <p
    contentEditable={true}
  >
    Hi, this is a test
  </p>
</div>
`;

exports[`<EditableContainer> snapshots render buttons correctly when set to basic 1`] = `
<div>
  <div
    id=""
  >
    <button
      className="bold"
      onClick={[Function]}
      type="button"
    >
      bold
    </button>
    <button
      className="italic"
      onClick={[Function]}
      type="button"
    >
      italic
    </button>
    <button
      className="underline"
      onClick={[Function]}
      type="button"
    >
      underline
    </button>
    <button
      className="strikeThrough"
      onClick={[Function]}
      type="button"
    >
      strikeThrough
    </button>
    <button
      className="createLink"
      onClick={[Function]}
      type="button"
    >
      createLink
    </button>
    <br />
    <button
      onClick={undefined}
      type="button"
    >
      Submit
    </button>
  </div>
  <p
    contentEditable={true}
  >
    Hi, this is a test
  </p>
</div>
`;

exports[`<EditableContainer> snapshots render buttons correctly when set to none 1`] = `
<div>
  <p
    contentEditable={true}
  >
    Hi, this is a test
  </p>
</div>
`;

exports[`<EditableContainer> snapshots render correctly when canEdit is set to false 1`] = `
<p>
  Hi, this is a test
</p>
`;

exports[`<EditableContainer> snapshots render correctly when canEdit is set to true 1`] = `
<div>
  <div
    id="buttonContainer"
  >
    <button
      className="bold"
      onClick={[Function]}
      type="button"
    >
      bold
    </button>
    <button
      className="italic"
      onClick={[Function]}
      type="button"
    >
      italic
    </button>
    <button
      className="underline"
      onClick={[Function]}
      type="button"
    >
      underline
    </button>
    <button
      className="heading"
      onClick={[Function]}
      type="button"
    >
      heading
    </button>
    <button
      className="fontSize"
      onClick={[Function]}
      type="button"
    >
      fontSize
    </button>
    <button
      className="insertOrderedList"
      onClick={[Function]}
      type="button"
    >
      insertOrderedList
    </button>
    <button
      className="insertUnorderedList"
      onClick={[Function]}
      type="button"
    >
      insertUnorderedList
    </button>
    <button
      className="justifyFull"
      onClick={[Function]}
      type="button"
    >
      justifyFull
    </button>
    <button
      className="justifyLeft"
      onClick={[Function]}
      type="button"
    >
      justifyLeft
    </button>
    <button
      className="justifyRight"
      onClick={[Function]}
      type="button"
    >
      justifyRight
    </button>
    <button
      className="justifyCenter"
      onClick={[Function]}
      type="button"
    >
      justifyCenter
    </button>
    <button
      className="strikeThrough"
      onClick={[Function]}
      type="button"
    >
      strikeThrough
    </button>
    <button
      className="subscript"
      onClick={[Function]}
      type="button"
    >
      subscript
    </button>
    <button
      className="superscript"
      onClick={[Function]}
      type="button"
    >
      superscript
    </button>
    <button
      className="insertImage"
      onClick={[Function]}
      type="button"
    >
      insertImage
    </button>
    <button
      className="createLink"
      onClick={[Function]}
      type="button"
    >
      createLink
    </button>
    <button
      className="unlink"
      onClick={[Function]}
      type="button"
    >
      unlink
    </button>
    <button
      className="hideFromPreview"
      onClick={[Function]}
      type="button"
    >
      hideFromPreview
    </button>
    <br />
    <button
      onClick={undefined}
      type="button"
    >
      Submit
    </button>
  </div>
  <p
    contentEditable={true}
  >
    Hi, this is a test
  </p>
</div>
`;
